### PR TITLE
feat: add finished scope to vacation

### PR DESCRIPTION
**Pull request by luccamps**
Solves #399

### CODE REVIEW

File: app/models/vacation.rb [hunk 1]
```diff
@@ -60,6 +60,12 @@ class Vacation < ApplicationRecord
     where("#{approver}_approver_id": nil)
   }
 
+  scope :finished, -> {
+    where(status: :approved)
```
Comment (adbatista) [inline app/models/vacation.rb:64]: what about using `approved` instead this where

File: spec/models/vacation_spec.rb [hunk 2]
```diff
@@ -151,6 +151,33 @@
...
+      expect(described_class.finished).to eq(ended_vacations)
+    end
+
+    it 'returns finished vacations sorted by most recent' do
+      ended_vacation.update(end_date: 2.days.ago)
+      sorted_ended_vacations = ended_vacations.push(ended_vacation)
+
+      expect(described_class.finished).to eq(sorted_ended_vacations)
+    end
+  end
```
Comment (adbatista) [inline spec/models/vacation_spec.rb:179]:
```suggestion
  describe '.finished' do
    let!(:ended_vacation1) { create(:vacation, :ended, end_date: 1.day.ago) }
    let!(:ended_vacation2) { create(:vacation, :ended, end_date: 2.days.ago) }
    let!(:ended_vacation3) { create(:vacation, :ended, end_date: 3.days.ago) }

    before do
      create(:vacation, :approved)
      create(:vacation, :pending)
      create(:vacation, :cancelled)
      create(:vacation, :denied)
      create(:vacation, :ongoing)
      create(:vacation, :scheduled)
      create(:vacation, :expired)
      create(:vacation, :valid)
    end
    
    it 'returns only finished vacations sorted by most recent' do
      expect(described_class.finished).to eq([ended_vacation1, ended_vacation2, ended_vacation3])
    end
  end
```

File: app/models/vacation.rb (hunk 1 above)
Comment (luccamps) [reply app/models/vacation.rb]: Way better, totally missed that example that used pending.

File: spec/models/vacation_spec.rb (hunk 2 above)
Comment (luccamps) [reply spec/models/vacation_spec.rb]: I thought that would look ugly but between two uglies better pick the lesser ugly :^). I have also removed that redundant test.

**Review by adbatista — APPROVED** [review]

**Review by adbatista — APPROVED** [review]

**Pull request merged**